index.js

```diff
@@ -48,5 +38,22 @@ module.exports = function (req, time) {
 		}
 	}
 
+	function connect() {
+		clear();
+
+		if (delays.socket !== undefined) {
+			// Abort the request if there is no activity on the socket for more
+			// than `delays.socket` milliseconds.
+			this.setTimeout(delays.socket, function socketTimeoutHandler() {
+				req.abort();
+				var e = new Error('Socket timed out on request' + host);
+				e.code = 'ESOCKETTIMEDOUT';
+				req.emit('error', e);
+			});
+		}
+	}
+
+	connect.isTimedOutListener = true;
```

Comment:
I'm not very happy with this but I have no better ideas, unfortunately symbols can't be used. Suggestions are welcome.